common/src/main/java/org/apache/comet/vector/CometPlainVector.java

```diff
@@ -111,7 +115,12 @@ public UTF8String getUTF8String(int rowId) {
       byte[] result = new byte[length];
       Platform.copyMemory(
           null, valueBufferAddress + offset, result, Platform.BYTE_ARRAY_OFFSET, length);
-      return UTF8String.fromBytes(result);
+
+      if (!isUuid) {
+        return UTF8String.fromBytes(result);
+      } else {
+        return UTF8String.fromString(convertToUuid(result).toString());
+      }
```

Comment:
For test, I think if it is possible to create a Parquet file with uuid column in unit test and read it? Due to this change the column should be read as uuid instead of string.

Reply:
I looked `makeParquetFileAllTypes`, seems I can only use parquet type `INT64`, `INT32`, `BOOLEAN`, `BINARY`, `FLOAT`, `DOUBLE`, `INT96`, `FIXED_LEN_BYTE_ARRAY`. It doesn't seem I can use a UUID logical type in a Parquet schema. When I did my local test using iceberg, I was using iceberg's `UUIDType`

Reply:
Hmm,

```
optional binary                   _13(ENUM);
```

I think we can specify physical type and logical type in the message, no? For example, above line specifies `binary` as physical type and `ENUM` as logical type.

Reply:
So I guess you can try

```
optional binary                   _13(UUID);
```

?

Reply:
I tried 
```
    val schemaStr =
      """
        |message root {
        |  required int32 id;
        |  required FIXED_LEN_BYTE_ARRAY(16) uuid (UUID);
        |}
      """.stripMargin
```
I got 
```
Fixed Binary size 36 does not match field type length 16
java.lang.IllegalArgumentException: Fixed Binary size 36 does not match field type length 16
	at org.apache.parquet.column.values.plain.FixedLenByteArrayPlainValuesWriter.writeBytes(FixedLenByteArrayPlainValuesWriter.java:56)
	at org.apache.parquet.column.impl.ColumnWriterBase.write(ColumnWriterBase.java:241)
	at org.apache.parquet.io.MessageColumnIO$MessageColumnIORecordConsumer.addBinary(MessageColumnIO.java:476)
	at org.apache.parquet.example.data.simple.BinaryValue.writeValue(BinaryValue.java:45)
	at org.apache.parquet.example.data.simple.SimpleGroup.writeValue(SimpleGroup.java:233)
	at org.apache.parquet.example.data.GroupWriter.writeGroup(GroupWriter.java:51)
	at org.apache.parquet.example.data.GroupWriter.write(GroupWriter.java:37)
	at org.apache.parquet.hadoop.example.GroupWriteSupport.write(GroupWriteSupport.java:85)
	at org.apache.parquet.hadoop.example.GroupWriteSupport.write(GroupWriteSupport.java:35)
	at org.apache.parquet.hadoop.InternalParquetRecordWriter.write(InternalParquetRecordWriter.java:138)
	at org.apache.parquet.hadoop.ParquetWriter.write(ParquetWriter.java:310)
	at org.apache.spark.sql.CometTestBase.$anonfun$createParquetFileWithIDAndUUID$1(CometTestBase.scala:482)
```

It has to be `FIXED_LEN_BYTE_ARRAY(16) uuid (UUID)`, otherwise I will get

```
UUID can only annotate FIXED_LEN_BYTE_ARRAY(16)
java.lang.IllegalStateException: UUID can only annotate FIXED_LEN_BYTE_ARRAY(16)
	at org.apache.parquet.Preconditions.checkState(Preconditions.java:196)
	at org.apache.parquet.schema.Types$BasePrimitiveBuilder$1.checkFixedPrimitiveType(Types.java:556)
```

Reply:
Might be something wrong with the UUID value. I will check

Reply:
I fixed the UUID data problem, but now I got `illegalParquetTypeError` from Spark. I don't think Spark supports Parquet's `UUID`. Iceberg maps `UUID` to `UTF8String`
```
Job aborted due to stage failure: Task 0 in stage 0.0 failed 1 times, most recent failure: Lost task 0.0 in stage 0.0 (TID 0) (192.168.50.140 executor driver): org.apache.spark.sql.AnalysisException: Illegal Parquet type: FIXED_LEN_BYTE_ARRAY (UUID).
	at org.apache.spark.sql.errors.QueryCompilationErrors$.illegalParquetTypeError(QueryCompilationErrors.scala:1762)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.illegalType$1(ParquetSchemaConverter.scala:206)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertPrimitiveField$2(ParquetSchemaConverter.scala:310)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertPrimitiveField(ParquetSchemaConverter.scala:224)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertField(ParquetSchemaConverter.scala:187)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertInternal$3(ParquetSchemaConverter.scala:147)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.$anonfun$convertInternal$3$adapted(ParquetSchemaConverter.scala:117)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:286)
	at scala.collection.immutable.Range.foreach(Range.scala:158)
	at scala.collection.TraversableLike.map(TraversableLike.scala:286)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:279)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convertInternal(ParquetSchemaConverter.scala:117)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.convert(ParquetSchemaConverter.scala:87)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat$.$anonfun$readSchemaFromFooter$2(ParquetFileFormat.scala:493)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat$.readSchemaFromFooter(ParquetFileFormat.scala:493)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat$.$anonfun$mergeSchemasInParallel$2(ParquetFileFormat.scala:473)
	at scala.collection.immutable.Stream.map(Stream.scala:418)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat$.$anonfun$mergeSchemasInParallel$1(ParquetFileFormat.scala:473)
	at org.apache.spark.sql.execution.datasources.parquet.ParquetFileFormat$.$anonfun$mergeSchemasInParallel$1$adapted(ParquetFileFormat.scala:464)
```

Reply:
Hmm, okay, we still use Spark ParquetToSparkSchemaConverter to convert Parquet schema to Spark schema? I thought that we may have some custom one in Comet.

Reply:
It failed [here](https://github.com/apache/datafusion-comet/blob/main/spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala#L370). It's Spark code, not Comet code yet. That's why it uses Spark's `ParquetToSparkSchemaConverter`

Reply:
The failed one is Spark reads Parquet file using its data source. Can you try to read it with Comet? If Comet scan doesn't use `ParquetToSparkSchemaConverter`, I think we won't hit the above error?